HW-1 네이버API로 영화 검색

chuuuul opened the pull request:
- 영화이름을 입력하고 검색버튼을 누르면 하단에 리스트로 영화 목록 출력
 - 목록에 있는 영화 아이템을 선택하면 웹브라우져로 이동
 - 앞으로 스터디 할 내용들(MVP, MVVM, DataBinding 등등 사용하지 않고 과제 하기)

chuuuul committed:
SearchActivity 생성

chuuuul committed:
Recyclerview import

chuuuul committed:
검색 액티비티 XML 작성

chuuuul committed:
설정 - 코드 스타일 ( kotlin ) 수정

chuuuul committed:
SearchActivity 이름 변경

chuuuul committed:
SearchActivity 폴더 이동

chuuuul committed:
영화 아이템 VO 생성

chuuuul committed:
영화 아이템 레이아웃 생성

chuuuul committed:
pubData를 pubDate로 이름 변경
Int형 변수에서 String형 변수로 변경

chuuuul committed:
userRating Float형으로 변경

chuuuul committed:
영화 아이템 생성

chuuuul committed:
Recyclerview Adapter, ViewHolder 생성

chuuuul committed:
패키지명 변경

chuuuul committed:
인터넷 사용 퍼미션 허용

chuuuul committed:
Load Error 이미지 추가

chuuuul committed:
레이아웃 업데이트

chuuuul committed:
Movie 리사이클러뷰 적용

chuuuul committed:
Recyclerview Adapter 클래스명 변경

 - SearchListAdapter -> MovieListAdapter
 - SearchViewHolder -> MovieViewHolder

chuuuul committed:
패키지명 수정

 - activity.activity -> activity.search

chuuuul committed:
item_movie 레이아웃 수정

 - 이미지 크기 0dp로 수정

chuuuul committed:
Glide의 PlaceHolder 이미지 추가

chuuuul committed:
코드 정리

 - PlaceHolder 추가
 - 반복되는 코드 제거 ( ?: 엘비스 연산자 )

chuuuul committed:
코드 스타일 변경

chuuuul committed:
Retrofit 임포트

chuuuul committed:
Gson 임포트

chuuuul committed:
Gson 버전 변경

chuuuul committed:
Import Gson - retrofit converter

chuuuul committed:
API Key 등록

chuuuul committed:
Retrofit 인터페이스 생성

chuuuul committed:
Retrofit Response VO 생성

chuuuul committed:
표현식 수정

 - url이 null이 아닌 "" 일 경우 No Image 표시하도록 변경

chuuuul committed:
SerializedName 추가

chuuuul committed:
Retrofit 인터페이스 작성

chuuuul committed:
문자열 처리 로직 추가

 - title에 검색한 키워드를 강조하는 <b> </b> 태그 제거

chuuuul committed:
이미지뷰 높이 수정

chuuuul committed:
검색 기능 구현 ( Retrofit2 )

chuuuul committed:
package 변경

chuuuul committed:
영화 클릭 이벤트 구현

 - 영화 클릭 시 웹페이지로 이동

chuuuul committed:
검색창 한줄로 제한

chuuuul committed:
EditText 엔터 기능 구현

 - 엔터 버튼 누를 시 검색 버튼 클릭

chuuuul committed:
람다식 수정

chuuuul committed:
EditText 버그 수정

 - 검색창 안지워지는 버그 수정

chuuuul committed:
image변수의 Nullable 제거

 - API Document에 '이미지가 있는경우에만 나타난다'를 이미지가 없을 때 Null 이라는 의미로 잘못 이해함

chuuuul committed:
if 조건 변경

 - 가독성을 위해 부등호로 변경

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/Activity/MovieVO.kt:
```diff
@@ -0,0 +1,11 @@
+
+data class MovieVO(
+        val image : String?,
+        val title : String,
```
안녕하세요 잘부탁드립니다.
궁금해서 여쭤봅니다.
어떤건 nullable이고 어떤건 non-null인데 기준이 혹시 있으신건가요?

chuuuul replied:
네 잘부탁드립니다~ ㅎㅎ

그부분 제가 잘못해석 해서

최근 커밋(ede3f99)에서 수정했습니다!

Non - null로 해야되는게 맞는것 같습니다.

//////
image변수의 Nullable 제거 

 - API Document에 '이미지가 있는경우에만 나타난다'를 이미지가 없을 때 Null 이라는 의미로 잘못 이해함

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/res/layout/item_movie.xml:
```diff
@@ -7,7 +7,7 @@
         android:id="@+id/imageView_movie_Image"
         android:layout_width="0dp"
-        android:layout_height="200dp"
+        android:layout_height="250dp"
```
혹시 이미지뷰 높이를 고정으로 주신 이유가 있을까요?

chuuuul replied:
포스터 이미지를 넣기위해 임의로 지정했습니다. 가로 길이에 비례해서 높이를 지정하고 싶은데 방법을 몰라서 직접 잡았습니다. 해상도에 따른 문제라면 dimens.xml으로 관리 해야 될 것 같긴 한데, 다른 좋은 방법이 있으면 공유해주세요!

SageHabitus replied:
저같은 경우는 실무에서 adjustviewbounds를 많이 쓰고 있습니다.
이 속성에 대해서 한번 알아보시면 도움이 되실 것 같습니다!

chuuuul replied:
7ec458c
수정했습니다! 찾아봤는데 말씀하신 adjustviewbounds="true"는 1:1 비율로 고정되는거여서 제가 구현하고싶은 layout_constraintDimensionRatio="1:1.4" 를 사용해서 동적으로 비율 구현 했습니다. 저도 높이 고정로 잡은게 마음에 걸렸는데 도와주셔서 감사합니다!

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/activity/SearchActivity.kt:
```diff
@@ -11,6 +11,23 @@ class SearchActivity : AppCompatActivity() {
+    private fun setRecyclerview() {
+        val gridLayoutManager = GridLayoutManager(this, 2, GridLayoutManager.VERTICAL, false)
+
+        // * Test
```
*Test 주석의 의미는 뭔지 여쭤봐도 될까요~?
긴 문자열은 따로 모아서 한곳에 두면 좋을 것 같은데 어떻게 생각하시는지 궁금합니다.

chuuuul replied:
Test 주석은 제가 Retrofit으로 데이터 가져오기 전에 RecyclerView에 Adapter로 데이터가 잘 들어가는지 확인 하려고 만든 테스트 코드입니다!

긴 문자열은 따로 관리 해야하는것에 동의합니다 ㅎㅎ 다만 테스트용도로 잠깐 사용하고 이후 다른 커밋에서 삭제할 용도로 만들었기 때문에 따로 관리를 안했어요!

최종 PR 내용에서는 삭제되었습니다!

chuuuul committed:
주석 수정

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/res/layout/activity_search.xml:
```diff
@@ -6,13 +6,39 @@
+        android:layout_marginStart="16dp"
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:hint="검색 할 영화를 입력하세요"
```
저도 버릇들여야 하지만 hint나 text 같은 경우 string.xml에 리소스를 따로 모아서 관리하는 것이 유지보수에 매우 도움 된다고 알고 있습니다.(다국어지원할때도) 이번 스터디때 이런 리소스들은 string.xml을 적극 이용하면 매우 좋을 것 같습니다.

chuuuul replied:
585198a
감사합니다. 수정했습니다.
string.xml에 관리 해야하는건 알고 있는데 실행해본적이 없었네요. 이번 기회에 성실하게 관리해보겠습니다 ㅎㅎ

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/build.gradle:
```diff
@@ -34,4 +37,19 @@ dependencies {
+
+    // Retrofit 2
+    // Retrofit requires at minimum Java 8+ or Android API 21+.
+    implementation 'com.squareup.retrofit2:retrofit:2.6.1'
```
gradle 버전같은 경우에 주요 라이브러리들은 따로 모아서 ext. 형태로 관리하곤 하는데 이건 명확히 어떻게 하는게 좋다라는 것을 알고있는 것이 아니라 이 부분에 대해서도 한번 알아보시면 좋을 것 같습니다.

chuuuul replied:
208c9b7
감사합니다. 수정했습니다.
덕분에 알아갑니다 ㅎㅎ

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+
+    private fun setMovieImage(imageView: ImageView, imageUrl: String?) {
+
+        val url = if (imageUrl == "") NO_IMAGE_URL else imageUrl
```
== "" 보다는 isNullOrEmpty, isEmpty, isBlankOrEmpty 또는 String.확장함수 형태로 String.hasText() : Boolean 형태로 편하게 쓸 수도 있을 것 같습니다.

chuuuul replied:
dd5d697
감사합니다. 수정했습니다!

chuuuul replied:
dd5d697
감사합니다. 수정했습니다!

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+        return string.replace("<b>", "").replace("</b>", "")
+    }
+
+    private fun openWebPage(urlString: String) {
```
어차피 받는 값은 :String으로 명시가 돼있는데 굳이 urlString으로 중복선언하지 않아도 될 것 같습니다.

chuuuul replied:
이해를 못했습니다.. urlString이라고 변수명을 지은것에 대해서 말씀하시는게 맞나요?

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+class SearchActivity : AppCompatActivity(), Callback<MovieResponseVO> {
+
+    private val MOVIE_SEARCH_API_URL = "https://openapi.naver.com"
+    private var movies = ArrayList<MovieVO>()
```
DTO와 VO가 미묘한 차이가 있지만 거의 같습니다.
명명을 MovieVO로 하신 것에 대해서 혹시 개인적인 이유가 어떤것인지 단순히 궁금해서 코멘트 남깁니다

chuuuul replied:
감사합니다.
딱히 거기까진 생각을 안해봤습니다... 평소에는 Model이라는 단어를 많이 쓰는데 다른 사람의 코드 중에 VO라는 단어를 보고 나서 저도 VO라고 지었습니다.

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+        var isValid = true
+        var errorString = ""
+
+        if (keyword == "") {
```
개인적인 생각으로는 isEmpty가 더 자연스러워 보입니다.

chuuuul replied:
dd5d697
감사합니다. 수정했습니다!

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+        }
+
+        if (!isValid)
+            Toast.makeText(this, errorString, Toast.LENGTH_SHORT).show()
```
anko toast 에 대해서 한번 알아보시면 도움이 되실 것 같습니다. toast("") or longToast("") 형태로 아주 간결하게 표현이 가능합니다.

chuuuul replied:
꿀팁 감사합니다!
anko 찾아보니 deprecated 됐다고 하네요!

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+            } else {
+                movies.clear()
+                recyclerview_search_movieList.adapter?.notifyDataSetChanged()
+                Toast.makeText(this, "검색 결과가 없습니다.", Toast.LENGTH_SHORT).show()
```
string.xml에 대해서 정리하면 좋을 것 같습니다

chuuuul replied:
585198a
감사합니다 수정했습니다!

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/res/layout/activity_search.xml:
```diff
@@ -0,0 +1,46 @@
+    tools:context=".activity.search.SearchActivity">
+
+    <EditText
+        android:id="@+id/editText_search_searchName"
```
view를 명명하실때 et또는 edt_ <-- 이런식으로 줄여서 하는것이 어떨까 합니다.
textview는 tv또는 txv button은 btn 등등..
명확한 명명규칙만 있다면 줄여도 엑티비티에서 코틀린 익스텐션으로 코드를 가져올때 명확하게 코딩 할 수 있다고 봅니다.

chuuuul replied:
e6d2ba3 , c8d983c
감사합니다. 수정했습니다.

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/res/layout/activity_search.xml:
```diff
@@ -0,0 +1,46 @@
+        android:layout_marginTop="16dp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
```
bias 0.0을 혹시 명시하신 이유가 무엇인지 궁금합니다

chuuuul replied:
9d4c1d2
수정했습니다. 감사합니다!
일부러 지정한건 아니었고 어쩌다가 변경된것같습니다~

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/app/src/main/res/layout/item_movie.xml:
```diff
@@ -0,0 +1,58 @@
+        android:layout_height="wrap_content"
+        android:layout_marginTop="8dp"
+        android:textAlignment="center"
+        app:layout_constraintEnd_toEndOf="@+id/imageView_movie_Image"
```
imageview또한 명명은 iv_blahblah 또는 img_blahblah 등등이 좋을 것 같습니다

chuuuul replied:
e6d2ba3 , c8d983c
감사합니다. 수정했습니다.

SageHabitus commented on 2005/chuuuul/ArchitectureStudy/gradle.properties:
```diff
@@ -19,3 +19,5 @@ android.useAndroidX=true
 # Kotlin code style for this project: "official" or "obsolete":
 kotlin.code.style=official
+Naver_Client_Id="QKFaoQGF6ahDKBFnyljx"
+Naver_Client_Secret="qk5KsJir31"
```
저같은 경우는 이런 키 같은 경우에 string.xml이나 따로 Constants이름의 Object 를 만들어서 관리합니다만(정석인지는 모르겠습니다) 혹시 gradle.properties안에 명시하신 이유가 궁금합니다.

chuuuul replied:
일반적으로 API키 노출을 최대한 피해 string.xml나 코드에 직접 쓰기보단, 보안을 위해 숨긴다고 합니다. 그 중 한 방법이 gradle.properties 에 넣는 방법이구 다른 여러 방식이 있었습니다. 

API키가 노출되면 악의적으로 사용되기도 할테니까요! 오픈소스같이 프로젝트가 통짜로 공개되는 경우에는 local.properties에 API Key를 저장해고 local.properties는 깃에 올리지 않는 방식으로 관리해서 다른사람들이 API Key를 알 수 없게끔 하는 것으로 알고 있습니다.

다른방법도 많이 찾아보시면 도움이 되실것 같습니다 ㅎㅎ

관련 된 2004기 NukeOlaf님의 포스팅입니다!
https://salix97.tistory.com/236

SageHabitus replied:
감사합니다

chuuuul committed:
ImageView 높이 동적으로 수정

chuuuul committed:
내용 없는 String 조건 표현식 변경

 - if( string == "" ) -> if( string.isEmpty() )

chuuuul committed:
잘못 잡혀있던 bias 수정

 - layout_constraintHorizontal_bias 0.0 -> 제거
 - layout_constraintVertical_bias 0.0 -> 제거

chuuuul committed:
View 명명 변경

 - button : btn
 - textView : tv
 - imageView : img

chuuuul committed:
View 명명 변경

 - et -> edt

chuuuul committed:
문자열 관리

 - 문자열 관리 : values/strings.xml

chuuuul committed:
라이브러리 버전관리

 - project build.gradle의 ext에서 관리

chuuuul committed:
공백문자 입력 방지

 - isEmpty를 isBlank로 변경

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+class MovieListAdapter(private val movies: ArrayList<MovieVO>) :
+    RecyclerView.Adapter<MovieViewHolder>() {
+
+    private lateinit var context: Context
```
이 변수는 불필요 합니다.
제거 해보고 에러가 나는부분 수정해보세요
잘 안된다면 질문해주세요

chuuuul replied:
aebb6f5
감사합니다. 수정했습니다!
항상 해오던 습관이었는데, 이번에 잘 배웠습니다.

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
+        holder.itemView.tv_movie_title.text = removeMarkupTag(movies[position].title)
+        holder.itemView.tv_movie_pubDate.text = movies[position].pubDate
+
+        holder.itemView.ratingBar_movie_rating.rating = movies[position].userRating
+
+        setMovieImage(holder.itemView.img_movie_Image, movies[position].image)
+
+        holder.itemView.setOnClickListener {
+            openWebPage(movies[position].link)
+        }
```
movies[position]가 중복되는데 이 부분은 val로 빼서 사용하면 됩니다

chuuuul replied:
9e91257
감사합니다. 수정했습니다!

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+import com.example.architecture.vo.MovieVO
+import kotlinx.android.synthetic.main.item_movie.view.*
+
+class MovieListAdapter(private val movies: ArrayList<MovieVO>) :
```
list를 외부에서 받지 마시고 내부에서 가지고 있게 해주세요
내부에서 list객체를 별도로 관리하고 데이터만 넣어주도록 해주세요

chuuuul replied:
eadd3c2
감사합니다! 수정했습니다.

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/MovieListAdapter.kt:
```diff
@@ -0,0 +1,68 @@
+
+        setMovieImage(holder.itemView.img_movie_Image, movies[position].image)
+
+        holder.itemView.setOnClickListener {
```
ClickListener는 한번만 등록해주면 되는 작업이라서 
onCreateViewHolder에서 추가해주면 됩니다

chuuuul replied:
감사합니다!

말씀해주신것 고민해봤습니다! 그런데 onCreateViewHolder에 아이템의 Position값을 얻어 올 수가 없어서 ClickListener에서 영화의 링크를 가져올 수 가 없네요.. 

어떻게 링크값을 가져올 수 있을까요?

sjjeong replied:
ViewHolder에서 adapterPosition을 통해서 가져올 수 있습니다.

chuuuul replied:
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MovieViewHolder {

        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_movie, parent, false)

        view.setOnClickListener {
            openWebPage(it.context, movies[MovieViewHolder(it).adapterPosition].link )
        }

        return MovieViewHolder(view)
    }

이렇게 짰는데 MovieViewHolder(it).adapterPosition 값이 계속 -1이 나와서 에러가 납니다.

MovieViewHolder(it).position은 Deprecated 되었구
MovieViewHolder(it).layoutPosition도 -1값이 나와서 사용 할 수 없었습니다

sjjeong replied:
@chuuuul 
왜냐면 return해준 MovieViewHolder와 실제 setOnClickListener 안에 있는 MovieViewHolder가 다른 객체이기 때문이죠
같은 객체로 하게 해줘야 합니다.

chuuuul replied:
95ca737
아하. 이해했습니다! 정말 감사합니다!

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+
+class SearchActivity : AppCompatActivity(), Callback<MovieResponseVO> {
+
+    private val MOVIE_SEARCH_API_URL = "https://openapi.naver.com"
```
상수는 companion object를 만들어서 관리하는게 좋습니다

chuuuul replied:
5b0c6f4
감사합니다. 수정했습니다!

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+        }
+
+        // * 키보드 엔터 클릭 시 검색
+        edt_search_searchName.setOnKeyListener { _, keyCode, _ ->
```
setOnEditorActionListener로 하는게 더 좋을 것 같습니다.

xml에서 EditText에 아래 옵션을 추가해서 작업해주세요
```xml
android:imeOptions="actionSearch"
```

chuuuul replied:
b705f94, 57a7057
감사합니다. 수정했습니다!

세세한 부분까지 리뷰해주셔서 감사합니다!
덕분에 디테일까지 알아갑니다.

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+        }
+    }
+
+    private fun isValidKeyword(keyword: String): Boolean {
```
이렇게 하면 더 간략하게 만들 수 있습니다.
```kotlin
private fun isValidKeyword(keyword: String): Boolean {
    if (keyword.isBlank()) {
        Toast.makeText(this, getString(R.string.empty_keyword), Toast.LENGTH_SHORT).show()
        return false
    }
    return true
}
```

chuuuul replied:
fdd8f2b
감사합니다. 수정했습니다!
처음 작성 할 때 검색어 유효 검사할 항목이 많을 줄 알고 그렇게 구현 했었는데, Blank검사만 하면 되는거여서 불필요한 코드가 됐었네요~

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+
+    private fun setRecyclerview() {
+
+        val gridLayoutManager = GridLayoutManager(this, 2, GridLayoutManager.VERTICAL, false)
```
LayoutManager를 xml에서 설정할 수 있습니다

chuuuul replied:
561e9fb
감사합니다! 수정했습니다!

sjjeong commented on 2005/chuuuul/ArchitectureStudy/app/src/main/java/com/example/architecture/activity/search/SearchActivity.kt:
```diff
@@ -0,0 +1,118 @@
+
+            if (result.total > 0) {
+                movies.clear()
+                movies.addAll(response.body()?.movies!!)
```
result를 사용하면 됩니다.
```kotlin
movies.addAll(result.movies)
```

chuuuul replied:
08aa60c
감사합니다. 수정했습니다!

chuuuul committed:
중복 코드 제거

 - movies[position]을 val 변수로 빼서 사용

chuuuul committed:
Recycler View에서 Context 변수 제거

 - movies[position]을 val 변수로 빼서 사용

chuuuul committed:
반복되는 코드 수정

 - if문 대신 스코프 사용
 - result를 it으로 대체

chuuuul committed:
Recycler View의 List 객체를 분리해서 관리하도록 수정

 - Adapter의 파라미터로 List를 받던 것을 새로운 메서드를 생성해서 아이템을 받도록 수정

chuuuul committed:
상수 변수 관리 클래스 생성

 - 상수는 Companion object를 이용해서 관리하는게 좋다.

chuuuul committed:
EditText의 키보드 엔터 아이콘을 검색 아이콘으로 변경

 - android:imeOptions="actionSearch" 사용
 - if (keyCode == KeyEvent.KEYCODE_SEARCH) { ... }

chuuuul committed:
isValidKeyword 메서드 코드 간략화

chuuuul committed:
EditText의 키보드 엔터 아이콘을 검색 아이콘으로 변경

chuuuul committed:
LayoutManager를 코드에서 Xml로 이동

chuuuul committed:
키보드 엔터 이벤트를 setOnEditorActionListener로 수정

chuuuul committed:
상수저장 위치 변경

chuuuul committed:
Recycler View의 onClickListener 수정

 - Recycler View의 onClickListener를 onCreateViewHolder에서 관리 (+1 squashed commits)

Squashed commits:

[54e2b9ee7] Recycler View의 onClickListener 수정

 - Recycler View의 onClickListener를 onCreateViewHolder에서 관리

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
